### ISSUE TITLE
how do you get the ground truth scene coordinates?

### ISSUE THREAD

**Issue by Guan2014** [vislearn/esac#4]
Thanks for sharing your code. I want to test the accuracy of predicted scene coordinates but the ground truth for the test data was not provided. I used the depth images, camere focal length 525 and the ground truth poses to tranform the pixels to their corresponding 3D coordinates, by which I got the ground truth scene coordinates for the test data. Howerer, the coordinates errors I got is large. Could you tell me how you get the ground truth scene coordinates?

**Comment by ebrach** [comment]
Hi,
in principle your description is correct. Details like the focal length depend on the specific data set. Also be aware that the scene coordinate output is sub-sampled by a factor of 8. That is, you should calculate ground truth scene coordinates for pixel positions (4, 4); (12, 4); (20, 4) and so on. 

For reference about the conversion from scene coordinates to camera coordinates to pixel positions, you can have a look at the Aachen dataset script: https://github.com/vislearn/esac/blob/aacaa3b58a2d993c8264f2d5e4d4294a41a84e9a/datasets/setup_aachen.py#L186-L198
You have to do the transformation in reversed direction, i.e. from pixel positions to scene coordinates.

I would suggest to try your implementation on the training set, where ground truth scene coordinates exist for a comparison. Probably it is something trivial, like having to invert the pose transformation. Good luck ;-)

**Comment by Guan2014** [comment]
@ebrach  Thanks for you reply. I find that the invalid points in depth images of 7Scenes are not consistent with the invalid points in the ground truth scene coordinates you provided. I wonder if you use a rendered depth image as the issue page  [https://github.com/vislearn/LessMore/issues/13](url) ?

**Comment by ebrach** [comment]
Good catch. Yes, the ground truth scene coordinate that we provide were generated from rendered depth maps. I can provide a link to our depth maps for the 7Scenes training set: 
https://cloudstore.zih.tu-dresden.de/index.php/s/IS7AsAUioKKX7JZ

Note that rendered vs. Kinect depth would not explain large differences between generated scene coordinates, as the depth maps are quite similar. But also note, that the original 7Scenes depth maps are not registered to the RGB sensor. I can provide a link to registered 7Scenes depth maps that we used in the past (the registration is not perfect, but ok):
https://cloudstore.zih.tu-dresden.de/index.php/s/xgRx2MbIdZ6db6n

Hope that helps to debug your issue!

Best,
Eric

**Comment by Guan2014** [comment]
That helps a lot and thank you very much!
